src/pages/index.js

```diff
@@ -85,125 +141,133 @@ const cardUrlInput = addCardFormElement.querySelector(".modal__input_type_url");
+    .catch(console.error);
 }
 
-// function handleEscape(e) {
-//   if (e.key === "Escape") {
-//     const openModal = document.querySelector(".modal_opened");
-//     closeModal(openModal);
-//   }
-// }
+function handleAvatarFormSubmit(data) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(data.url)
+    .then((res) => {
+      userInfo.setUserAvatar(res);
+      editAvatarPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      editAvatarPopup.renderLoading(false);
+    });
+}
 
-// // function handlePopupClose(evt) {
-// //   if (
-// //     evt.target === evt.currentTarget ||
-// //     evt.target.classList.contains("modal__close")
-// //   ) {
-// //     closeModal(evt.currentTarget);
-// //   }
-// }
+const deleteCardPopup = new PopupDeleteCard({
+  popupSelector: "#delete-card-modal",
+});
+deleteCardPopup.setEventListeners();
 
 // Profile Form
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-function handleProfileformSubmit(userData) {
-  profilePopupForm.close();
-  userInfo.setUserInfo(userData);
-}
+// function handleProfileformSubmit(userData) {
+//   // profilePopupForm.close();
+//   // userInfo.setUserInfo(userData);
+//   api
+//     .updateProfileInfo({
+//       name: userData.title,
+//       description: userData.description,
+//     })
+//     .then((updatedUserData) => {
+//       userInfo.setUserInfo({
+//         title: updatedUserData.name,
+//         description: updatedUserData.about,
+//       });
+//       profilePopup.close();
+//     })
+//     .catch(console.error);
+// }
 
 // Event Listeners
 
-// remove submission
-// profileEditForm.addEventListener("submit", handleProfileEditSubmit);
-// addCardFormElement.addEventListener("submit", handleAddCardSubmit);
-
 profileEditButton.addEventListener("click", () => {
   const { title, description } = userInfo.getUserInfo();
   profileTitleInput.value = title;
   profileDescriptionInput.value = description;
   editProfilePopup.open();
 });
+api.getUserInfo();
 
 addNewCardButton.addEventListener("click", () => addCardPopup.open());
 
-// For loop that inserts a card
-
-// initialCards.forEach((cardData) => renderCard(cardData, cardListElement));
-
 // Enbale Validation
 
 const addCardFormValidator = new FormValidator(config, addCardFormElement);
 addCardFormValidator.enableValidation();
 
 const editProfileFormValidator = new FormValidator(config, profileEditForm);
 editProfileFormValidator.enableValidation();
+
+function handleDeleteCard(cardId, card) {
+  deleteCardPopup.open();
+  deleteCardPopup.handleDeleteConfirm(() => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch(console.error);
+  });
+}
+
+function handleLikeCard(cardId, cardElement) {
+  if (!cardElement._isLiked) {
```

Comment:
`_isLiked`

* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.